pkg/backupdriver/backup_repository.go

```diff
@@ -46,19 +47,53 @@ func ClaimBackupRepository(ctx context.Context,
 		4. If there are no pre-existing BRC, create one. Wait for it to be associated with
 			a BR
 	*/
-	var backupRepository *backupdriverv1.BackupRepository
 	var backupRepositoryClaim *backupdriverv1.BackupRepositoryClaim
 	brcList, err := backupdriverV1Client.BackupRepositoryClaims(ns).List(metav1.ListOptions{})
 	if err != nil {
 		return nil, errors.Errorf("Failed to list backup repository claims from the %v namespace", ns)
 	}
 	if len(brcList.Items) > 0 {
+		logger.Infof("Found %d BackupRepositoryClaims", len(brcList.Items))
 		// Pre-existing BackupRepositoryClaims found.
 		// Assuming only the first one, in theory multiple backup repo claim could exist.
 		repositoryClaimFound := brcList.Items[0]
 		if repositoryClaimFound.BackupRepository != "" {
```

Comment:
What you do in this `if` block is more like that of `if repositoryClaimFound.BackupRepository == ""`.

BTW, get rid of the hardcoded first-item-only logic. Instead, traversing the list until find a qualified BackupRepository  claimed by any matching BackupRepositoryClaim.

Additionally, is it possible to have a BRC that has the BackupRepository field unset? If yes, that would be a bug in BackupDriver controller. Basically, BackupDriver controller will always be watching the creation of BackupRepositoryClaim and create BackupRepository accordingly.

So, `if repositoryClaimFound.BackupRepository == ""`, then we just need to skip it since BackupDriver controller will take care of it. If there is no qualified BRC, then create a new BRC CR and wait until its BR field is populated. the util function, ClaimBackupRepository don't have to explicitly create BR resource.

Reply:
Yes, BackupDriver should be creating BR when it sees a BRC.  I'm waiting for the create BR util function in this PR before checking in my changes.

Reply:
handled this in the latest change.